Cumul ch (#2)
avaldivi6/ModernDive_book: 07-sampling.Rmd

```{r sampling-exercise-1, echo=FALSE, fig.cap="A bowl with red and white balls.", purl=FALSE, out.width = "95%", purl=FALSE}
include_graphics("images/sampling/balls/sampling_bowl_1.jpg")
```

### The proportion of red balls in the bowl {#population-proportion}

To do this we can remove each ball individually, count the number of red balls, and divide the number of red balls by the total number of balls. The bowl seen in Figure \@ref(fig:sampling-exercise-1) is represented virtually by the data frame `bowl` included in the `moderndive` package. The first ten rows are shown here for illustration purposes:

```{r}
bowl
```

> The  `bowl` has `r num_balls` rows representing the `r num_balls` rows ball in the bowl shown in Figure \@ref(fig:sampling-exercise-1). You can view and scroll through the entire contents of the `bowl` in RStudio's data viewer by running `View(bowl)`. The first variable `ball_ID` is used as an *identification variable* as discussed in Subsection \@ref(identification-vs-measurement-variables); none of the balls in the actual bowl are marked with numbers. The second variable `color` indicates whether a particular virtual ball is red or white. We compute the proportion of red balls in the bowl using the `dplyr` data wrangling verbs presented in Chapter \@ref(wrangling). First, for each of the balls, we identify if it is red or not using a test for equality with `==`. We then create a new Boolean variable `is_red` using the `mutate()` function from Section \@ref(mutate):

Author: ```suggestion
The  `bowl` has `r num_balls` rows representing the `r num_balls` balls in the bowl shown in Figure \@ref(fig:sampling-exercise-1). You can view and scroll through the entire contents of the `bowl` in RStudio's data viewer by running `View(bowl)`. The first variable `ball_ID` is used as an *identification variable* as discussed in Subsection \@ref(identification-vs-measurement-variables); none of the balls in the actual bowl are marked with numbers. The second variable `color` indicates whether a particular virtual ball is red or white. We compute the proportion of red balls in the bowl using the `dplyr` data wrangling verbs presented in Chapter \@ref(wrangling). First, for each of the balls, we identify if it is red or not using a test for equality with `==`. We then create a new Boolean variable `is_red` using the `mutate()` function from Section \@ref(mutate):
```

  virtual_samples_25 <- read_rds("rds/virtual_samples_25.rds")
}
virtual_prop_red_25 <- virtual_samples_25 |>

>   summarize(prop_red = mean(color == "red"), n =n())

Author: ```suggestion
  summarize(prop_red = mean(color == "red"), n = n())
```

  virtual_samples_50 <- read_rds("rds/virtual_samples_50.rds")
}
virtual_prop_red_50 <- virtual_samples_50 |>

>   summarize(prop_red = mean(color == "red"), n =n())

Author: ```suggestion
  summarize(prop_red = mean(color == "red"), n = n())
```

  virtual_samples_100 <- read_rds("rds/virtual_samples_100.rds")
}
virtual_prop_red_100 <- virtual_samples_100 |>

>   summarize(prop_red = mean(color == "red"), n =n())

Author: ```suggestion
  summarize(prop_red = mean(color == "red"), n = n())
```

The first set of terms and notations relate to **populations**:

1. A **population** or **study population** is a collection of all individuals or observations that we are interested in. 
    - In our sampling activities, the **population** is the collection of balls in the bowl shown in Figure \@ref(fig:sampling-exercise-1).
1. If the size of the population is known, as in the bowl activity, we may use the upper-case letter $N$ to represent this size. 
    - For our sampling activities, for example, we use the virtual `bowl` to determine that the population size, the total number of balls in the bowl, was $N$ = `r num_balls`. 
    - For more general settings, it is not uncommon to deal with situations where the size of the population is in the thousands, millions, or even more. In real-life problems, the size of the population is often unknown and for some theoretical populations, we could deal with an infinite number of outcomes.
1. A **population parameter** or simply **parameter** is a numerical summary, a number, that represents some characteristic of the population. 

>     - In our sampling activities, the *population proportion*, denote it by the letter $p$, is the *parameter* of interest. In subsection \@ref(population-proportion) we calculated this value, $p$=`r prop_red`. 

Author: ```suggestion
    - In our sampling activities, the *population proportion*, denoted by the letter $p$, is the *parameter* of interest. In subsection \@ref(population-proportion) we calculated this value, $p$=`r prop_red`. 
```

> ### Sampling distributions, the expected value and the standard error {#sampling-definitions}

Author: ```suggestion
### Sampling distributions, the expected value, and the standard error {#sampling-definitions}
```

- When the sample size used is 25 (left histogram) it is somewhat likely to obtain sample proportions of red balls that are less than 0.2 or greater than 0.55. 
- When the sample size is 50 (middle histogram) it is no longer likely, although still possible, to get those extreme sample proportions.
- When the sample size is 100 (right histogram) it is very unlikely to get those sample proportions as we have not observed a single value that is that extreme in the simulation. 

This behavior is expected. Observe that as the sample size $n$ increases from 25 to 50 to 100, \index{sampling distributions!relationship to sample size} the variation of the sampling distribution decreases, and thus the values cluster more and more tightly around the population proportion. 
We quantify this variation using the standard deviation of sample proportions. Results are shown in Table \@ref(tab:comparing-n):

```{r comparing-n, echo=FALSE, purl=FALSE}

n <- c(25, 50, 100)
p <- prop_red

> sd.prop <- round(sqrt(p*(1-p)/n),3)

Author: ```suggestion
sd_prop <- round(sqrt(p * (1 - p) / n), 3)
```

comparing_n_table <- 

>   tibble(n,sd.prop) |> 

Author: ```suggestion
  tibble(n, sd_prop) |> 
```

>   rename(`Sample size used` = n, `Standard deviation of sample proportions (Standard Error)` = sd.prop)

Author: ```suggestion
  rename(`Sample size used` = n, `Standard deviation of sample proportions (Standard Error)` = sd_prop)
```

  summarize(prop_red = mean(color == "red"))
```

> The object `simul_X_bar_25` contains 10,000 sample proportions, each replicate was obtained following the experiment described earlier, so each sample proportion is a possible value of $\bar X$. 

Author: ```suggestion
The object `simul_X_bar_25` contains 10,000 sample proportions; each replicate was obtained following the experiment described earlier, so each sample proportion is a possible value of $\bar X$. 
```

>   summarise(mean_Xbar = mean(prop_red), SE_Xbar = sd(prop_red))

Author: ```suggestion
  summarize(mean_Xbar = mean(prop_red), SE_Xbar = sd(prop_red))
```

```

Similarly, if we obtain additional simulations with a sample size $n = 400$, the mean should still be $0.375$ and the standard error: $$SE_{\bar X} = \sqrt{\frac{p(1-p)}{n}} = \sqrt{\frac{0.375\cdot(1-0.375)}{400}} = 0.0242$$ These are the simulation results:

```{r}

> bowl |>

Author: @avaldivi6 I'm not sure why these got switched to `summarise` instead of `summarize`?

Learning about the expected value and standard error of the sample proportion, $\bar X$, is very useful. But this is also true for other essential characteristics of the distribution. 
In particular, histograms of these simulations will approximate visually really well the distribution of $\bar X$ for the different sample sizes. 
Figure \@ref(fig:comparing-sampling-distributions-for-clt) presents the histograms for all three simulations performed. The histograms reflect precisely what we have shown earlier about the mean and standard error of $\bar X$, the mean is equal to the population proportion $p=0.375$ and the standard error gets smaller for larger sample sizes following the equation $$SE_{\bar X} = \sqrt{\frac{p(1-p)}{n}}$$ 

```{r comparing-sampling-distributions-for-clt, echo=FALSE, fig.height=3, fig.cap="Histogram of the distribution of the sample proportion and the normal curve.", purl=FALSE}

# n = 25
if (!file.exists("rds/virtual_prop_25.rds")) {
  virtual_prop_25 <- bowl |>
    rep_slice_sample(n = 25, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n=n())
  write_rds(virtual_prop_25, "rds/virtual_prop_25.rds")
} else {

>   virtual_prop_25<- read_rds("rds/virtual_prop_25.rds")

Author: ```suggestion
  virtual_prop_25 <- read_rds("rds/virtual_prop_25.rds")
```

}

# n = 100
if (!file.exists("rds/virtual_prop_100.rds")) {
  virtual_prop_100 <- bowl |>
    rep_slice_sample(n = 100, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n=n())
  write_rds(virtual_prop_100, "rds/virtual_prop_100.rds")
} else {

>   virtual_prop_100<- read_rds("rds/virtual_prop_100.rds")

Author: ```suggestion
  virtual_prop_100 <- read_rds("rds/virtual_prop_100.rds")
```

}

# n = 400
if (!file.exists("rds/virtual_prop_400.rds")) {
  virtual_prop_400 <- bowl |>
    rep_slice_sample(n = 400, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n=n())
  write_rds(virtual_prop_400, "rds/virtual_prop_400.rds")
} else {

>   virtual_prop_400<- read_rds("rds/virtual_prop_400.rds")

Author: ```suggestion
  virtual_prop_400 <- read_rds("rds/virtual_prop_400.rds")
```

```

 
Moreover, the shape of the histogram provides a really good approximation of the shape of the sampling distribution of the sample proportion. If we were to draw a single smooth curve that connects the top of each bar on a histogram with each adjacent bar, the curves for each histogram would look similar to the ones presented in Figures
  \@ref(fig:sample-proportion-25-with-normal-pdf), \@ref(fig:sample-proportion-100-with-normal-pdf), and \@ref(fig:sample-proportion-400-with-normal-pdf). 

```{r sample-proportion-25-with-normal-pdf, echo=FALSE, fig.height=3, fig.cap="Histogram of the distribution of the sample proportion and the normal curve.", purl=FALSE}

> n = 25

Author: @avaldivi6 Can you change these to `sd_p` instead of `sd.p`, change the `=` to `<-` and the spacing to be `p <- 9/24` instead of `p<-9/24` to be consistent with the coding style we used throughout the book?

p=9/24
sd.p = sqrt(p*(1-p)/n)

if (!file.exists("rds/virtual_prop_25.rds")) {
  virtual_prop_25 <- bowl |>
    rep_slice_sample(n = 25, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n = n())
  write_rds(virtual_prop_25, "rds/virtual_prop_25.rds")
} else {

>   virtual_prop_25<- read_rds("rds/virtual_prop_25.rds")

Author: ```suggestion
  virtual_prop_25 <- read_rds("rds/virtual_prop_25.rds")
```

p=9/24
sd.p = sqrt(p*(1-p)/n)

if (!file.exists("rds/virtual_prop_100.rds")) {
  virtual_prop_100 <- bowl |>
    rep_slice_sample(n = 100, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n = n())
  write_rds(virtual_prop_100, "rds/virtual_prop_100.rds")
} else {

>   virtual_prop_100<- read_rds("rds/virtual_prop_100.rds")

Author: ```suggestion
  virtual_prop_100 <- read_rds("rds/virtual_prop_100.rds")
```

p=9/24
sd.p = sqrt(p*(1-p)/n)

if (!file.exists("rds/virtual_prop_400.rds")) {
  virtual_prop_400 <- bowl |>
    rep_slice_sample(n = 400, replace = TRUE, reps = 10000) |>
    summarize(prop_red = mean(color == "red"), n = n())
  write_rds(virtual_prop_400, "rds/virtual_prop_400.rds")
} else {

>   virtual_prop_400<- read_rds("rds/virtual_prop_400.rds")

Author: ```suggestion
  virtual_prop_400 <- read_rds("rds/virtual_prop_400.rds")
```
